### DISCUSSION TITLE
Is there a way to nest like in SASS

### DISCUSSION THREAD

**Discussion by SamuelBolduc** [cssinjs/jss#1545] [Q&A]
I'm using react-jss.

In SASS, I would do something like this:

```
.button {
  background: grey;

  &.active {
    background: orange;
  }
}
```

Is there a way to do the same in JSS? I tried this:

```
createUseStyles({
  button: {
    background: "grey",

    "&$active": {
       background: "orange",
    },
});
```

But when I want to add both classes to an element, Typescript complains that the `active` class doesn't exist.

**Comment by rtivital** [comment]
You need to define `active` separately for this to work:
```tsx
createUseStyles({
  active: {},
  button: {
    background: "grey",

    "&$active": {
       background: "orange",
    },
});
```

**Reply by SamuelBolduc** [reply]
That's what I've been doing but it seemed redundant a little :/ am I understanding it correctly that in all cases the classes need to be defined at the top level of the `createUseStyles()` object argument?

**Reply by rtivital** [reply]
Yes
